include/bout/indexoffset.hxx

```diff
@@ -58,6 +61,9 @@ struct IndexOffset {
       throw BoutException("Offset in y (%d) would go out of bounds at %d", dy, index.ind);
     }
 #endif
+    if (ny == 1) {
```

Comment:
Maybe this should really be `ASSERT3(dy <= ny)` to avoid the branch in production code?

Actually, maybe that condition is equivalent to the `CHECK` above (L60)?

Reply:
I think similar asserts in the other directions would be useful. That way we simplify production code, giving the compiler the best chance of understanding what we're trying to do.